### DISCUSSION TITLE
2.2.3 deps conclicts between celery and flask.

### DISCUSSION THREAD

**Discussion by nobody4t** [apache/airflow#27179] [Q&A]
### Apache Airflow version

Other Airflow 2 version (please specify below)

### What happened

I try to install airflow 2.2.3. But  I got errors like: 
```
ERROR: pip's dependency resolver does not currently take into account all the packages that are installed. This behaviour is the source of the following dependency conflicts.
celery 5.2.3 requires click<9.0,>=8.0.3, but you have click 7.1.2 which is incompatible.
```

airflow here requires celery at least 5.2.3 which requires click <9.0,>=8.0.3. But flask 1.1.4 requres 7.1.2.

### What you think should happen instead

No deps conflicts.

### How to reproduce

Just install airflow 2.2.3.

### Operating System

ubuntu 20.04

### Versions of Apache Airflow Providers

2.2.3

### Deployment

Other

### Deployment details

pip install airflow==2.2.3
python 3.10.6

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

**Comment by boring-cyborg** [comment]
Thanks for opening your first issue here! Be sure to follow the issue template!

**Comment by eladkal** [comment]
> airflow here requires celery at least 5.2.3

This is not the version in constraints.
https://raw.githubusercontent.com/apache/airflow/constraints-2.2.3/constraints-3.7.txt
Please follow documentation about best practice to install airflow
https://airflow.apache.org/docs/apache-airflow/stable/installation/installing-from-pypi.html

**Comment by nobody4t** [comment]
Close this. Because I found the ubuntu 22.04 uses python 3.10 which is not compatible with 2.2.3
